03_letters/hsy3418/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+// ByLetter implements sort.Interface for string[] based on the alphabetical order.
+type ByLetter []string
+
+func (a ByLetter) Len() int           { return len(a) }
+func (a ByLetter) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByLetter) Less(i, j int) bool { return a[i] < a[j] }
+
+//letters is a function returns a mapping of each letter to its frequency.
+func letters(s string) map[rune]int {
+	m := make(map[rune]int)
+	for _, runeValue := range s {
+		m[runeValue]++
+	}
+	return m
+}
+
+//sorLetters is a function returns a sorted slice of strings with elements
+func sortLetters(m map[rune]int) []string {
+	sortSlices := make(ByLetter, 0, len(m))
+	for key, val := range m {
+		sortSlices = append(sortSlices, fmt.Sprintf("%c:%d", key, val))
+	}
+	sort.Sort(sortSlices)
```

Comment:
`sort.Strings()` can be used to sort string slice. If `ByLetter` is created just for sorting, probably no need for creating this custom type.

Reply:
Please fix.